From 042fb75eda64b8114ec2563523afb41dfac2697f Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 01:50:48 -0400
Subject: [iommu] vt-d: Make domain_add_dev_info() take struct device

Message-id: <20140905015048.31297.53888.stgit@gir.stowe>
Patchwork-id: 91343
O-Subject: [RHEL7.1 PATCH 50/65] iommu/vt-d: Make domain_add_dev_info() take struct device
Bugzilla: 1129880 1087643
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1087643
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1129880
Upstream Status: 5913c9bf0efecfa869375af8b4619529d3464038

commit 5913c9bf0efecfa869375af8b4619529d3464038
Author: David Woodhouse <David.Woodhouse@intel.com>
Date:   Sun Mar 9 16:27:31 2014 -0700

    iommu/vt-d: Make domain_add_dev_info() take struct device

    Signed-off-by: David Woodhouse <David.Woodhouse@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/intel-iommu.c b/drivers/iommu/intel-iommu.c
index 66c2921..2333f36 100644
--- a/drivers/iommu/intel-iommu.c
+++ b/drivers/iommu/intel-iommu.c
@@ -2470,25 +2470,24 @@ static int identity_mapping(struct device *dev)
 }
 
 static int domain_add_dev_info(struct dmar_domain *domain,
-          struct pci_dev *pdev,
-          int translation)
+          struct device *dev, int translation)
 {
  struct dmar_domain *ndomain;
  struct intel_iommu *iommu;
  u8 bus, devfn;
  int ret;
 
- iommu = device_to_iommu(&pdev->dev, &bus, &devfn);
+ iommu = device_to_iommu(dev, &bus, &devfn);
  if (!iommu)
   return -ENODEV;
 
- ndomain = dmar_insert_dev_info(iommu, bus, devfn, &pdev->dev, domain);
+ ndomain = dmar_insert_dev_info(iommu, bus, devfn, dev, domain);
  if (ndomain != domain)
   return -EBUSY;
 
- ret = domain_context_mapping(domain, &pdev->dev, translation);
+ ret = domain_context_mapping(domain, dev, translation);
  if (ret) {
-  domain_remove_one_dev_info(domain, &pdev->dev);
+  domain_remove_one_dev_info(domain, dev);
   return ret;
  }
 
@@ -2607,7 +2606,7 @@ static int __init iommu_prepare_static_identity_mapping(int hw)
 
  for_each_pci_dev(pdev) {
   if (iommu_should_identity_map(&pdev->dev, 1)) {
-   ret = domain_add_dev_info(si_domain, pdev,
+   ret = domain_add_dev_info(si_domain, &pdev->dev,
           hw ? CONTEXT_TT_PASS_THROUGH :
         CONTEXT_TT_MULTI_LEVEL);
    if (ret) {
@@ -2940,7 +2939,7 @@ static int iommu_no_mapping(struct device *dev)
    */
   if (iommu_should_identity_map(&pdev->dev, 0)) {
    int ret;
-   ret = domain_add_dev_info(si_domain, pdev,
+   ret = domain_add_dev_info(si_domain, dev,
         hw_pass_through ?
         CONTEXT_TT_PASS_THROUGH :
         CONTEXT_TT_MULTI_LEVEL);
@@ -4201,7 +4200,7 @@ static int intel_iommu_attach_device(struct iommu_domain *domain,
   dmar_domain->agaw--;
  }
 
- return domain_add_dev_info(dmar_domain, pdev, CONTEXT_TT_MULTI_LEVEL);
+ return domain_add_dev_info(dmar_domain, dev, CONTEXT_TT_MULTI_LEVEL);
 }
 
 static void intel_iommu_detach_device(struct iommu_domain *domain,
-- 
1.7.1